lang/ast/src/exp/literal.rs

```diff
@@ -0,0 +1,112 @@
+use derivative::Derivative;
+use pretty::DocAllocator;
+
+use polarity_lang_miette_util::codespan::Span;
+use polarity_lang_printer::{Alloc, Builder, Precedence, Print, PrintCfg};
+
+use crate::{
+    ContainsMetaVars, Exp, FreeVars, HasSpan, HasType, MetaVar, Occurs, Shift, ShiftRange, Subst,
+    Substitutable, Zonk, ZonkError,
+    ctx::LevelCtx,
+    rename::{Rename, RenameCtx},
+};
+
+#[derive(Debug, Clone, PartialEq, Hash, Derivative)]
+pub struct Literal {
+    /// Source code location
+    pub span: Option<Span>,
+
+    /// The kind of literals with its concrete payload
+    pub kind: LiteralKind,
+
+    /// The type for literals gets resolved at lowering and will not change
+    #[derivative(PartialEq = "ignore", Hash = "ignore")]
+    pub inferred_type: Box<Exp>,
+}
+
+#[derive(Debug, Clone, PartialEq, Hash)]
+pub enum LiteralKind {
+    Str { original: String, unescaped: String },
```

Comment:
```suggestion
    String { original: String, unescaped: String },
```

is perhaps less confusing.